ios/tunnel/tunnel_api.go

```diff
@@ -241,8 +315,13 @@ type deviceLister interface {
 type manualPairingTunnelStart struct {
 }
 
-func (m manualPairingTunnelStart) StartTunnel(ctx context.Context, device ios.DeviceEntry, p PairRecordManager, version *semver.Version) (Tunnel, error) {
+func (m manualPairingTunnelStart) StartTunnel(ctx context.Context, device ios.DeviceEntry, p PairRecordManager, version *semver.Version, userspaceTUN bool) (Tunnel, error) {
 	if version.Major() >= 17 && version.Minor() >= 4 {
+		if userspaceTUN {
```

Comment:
that would be one place where it's easy to use separate interfaces for the two tunnel types. The `tunnelStarter` was only made for unit testing (and I just realized that they were just removed), but now we could have two different implementations for the tunnel types

Reply:
Yeah I am not a big fan of these Java/Mockito style whitebox unit tests that just ensure the right methods were called. Imo, if you have to use a mock, it's probably not worth testing it.